src/utils/__tests__/transactions.test.ts

```diff
@@ -59,4 +65,34 @@ describe('transactions', () => {
       expect(getQueuedTransactionCount(txPage)).toBe('1')
     })
   })
+
+  describe('getTxOrigin', () => {
+    it('should return undefined if no app is provided', () => {
+      expect(getTxOrigin()).toBe(undefined)
+    })
+
+    it('should return a stringified object with the app name and url', () => {
+      const app = {
+        url: 'https://test.com',
+        name: 'Test name',
+      } as SafeAppData
+
+      expect(getTxOrigin(app)).toBe('{"url":"https://test.com","name":"Test name"}')
+    })
+
+    it('should limit the origin to 200 characters with preference of the URL', () => {
+      const app = {
+        url: 'https://test.com/' + 'a'.repeat(160),
+        name: 'Test name',
+      } as SafeAppData
+
+      const result = getTxOrigin(app)
+
+      expect(result?.length).toBe(200)
+
+      expect(result).toBe(
+        '{"url":"https://test.com/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","name":"Tes"}',
+      )
+    })
+  })
```

Comment:
I would suggest adding another test case where the URL is long enough that it would also be cut-off and name would be empty

Reply:
I've added one.